File: client/src/types/workflow.ts
export interface Task {
  type: string;
  name: string;
  config: Record<string, any>;
  nextTasks: string[] | null;
  conditions: Record<string, string> | null;
  templateId?: string | null;
  parameters?: Record<string, any> | null;
  templateBased?: boolean;
  configOverrides?: Record<string, any> | null;
}

export interface TaskTemplate {
  id: string;
  name: string;
  description: string;
  type: string;
  category: string;
  defaultConfig: Record<string, any> | null;
  configSchema: Record<string, any> | null;
  createdBy: string;
  isActive: boolean;
  version: string;
  metadata: Record<string, any> | null;
}

export interface Workflow {
  id: string;
  name: string;
  description: string;
  createdBy: string;
  tasks: Task[];
  metadata: Record<string, any> | null;
}

export interface WorkflowExecutionInput {
  input: any;
} 

File: server/src/main/java/com/github/tanyonghe/flowforge/model/Workflow.java
package com.github.tanyonghe.flowforge.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.List;
import java.util.Map;

@Data
@Document(collection = "workflows")
public class Workflow {
    @Id
    private String id;
    private String name;
    private String description;
    private String createdBy;
    private List<Task> tasks;
    private Map<String, Object> metadata;

    @Data
    public static class Task {
        private String type;
        private String name;
        private Map<String, Object> config;
        private List<String> nextTasks;
        private Map<String, String> conditions;
        private String templateId;
        private Map<String, Object> parameters;
        private Boolean templateBased;
        private Map<String, Object> configOverrides;
    }

    public void setId(String id) {
        this.id = id;
    }
} 